Userland/Libraries/LibELF/DynamicObject.h

```diff
@@ -412,6 +419,32 @@ inline void DynamicObject::RelocationSection::for_each_relocation(F func) const
     });
 }
 
+template<typename F>
+inline void DynamicObject::for_each_relr_relocation(F f) const
+{
+    auto section = relr_relocation_section();
+    if (section.entry_count() == 0)
+        return;
+
+    auto* entries = reinterpret_cast<ElfW(Relr)*>(section.address().get());
+    auto base = base_address().get();
+    FlatPtr patch_addr = 0;
+    for (unsigned i = 0; i < section.entry_count(); ++i) {
```

Comment:
This method checks the entry count reported in the section without checking the actual size of the section. So maliciously formatted section could cause us to do weird stuff to the binary. Do the other iteration helpers have that kind of bounds check? If not :yakplus:, we should create an issue to track it.